### PR TITLE
Updating google providers to 2.9

### PR DESCRIPTION
**Pull request by marko7460**

### CODE REVIEW

File: examples/deploy_service/main.tf [hunk 1]
```diff
@@ -19,12 +19,12 @@ locals {
...
 
 provider "google" {
-  version = "~> 2.7.0"
+  version = "~> 2.9.0"
   region  = "${var.region}"
 }
 
 provider "google-beta" {
-  version = "~> 2.7.0"
+  version = "~> 2.9.0"
```
Comment (chrislovecnm) [inline examples/deploy_service/main.tf]: 2.9.1 is out ... any reason we do not use that??

Comment (morgante) [reply examples/deploy_service/main.tf]: We don't require 2.9.1, so there's no reason to require it. This matches 2.9.X.

**Comment by chrislovecnm** [comment]
@marko7460 

```
kubernetes nodes pool-02 has the expected taints
```

Your taint test failed.

**Comment by marko7460** [comment]
@chrislovecnm that test has been intermittently failing for a while. we removed it in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/171.

**Comment by aaron-lane** [comment]
Ah, we did not remove the test correctly. The assertion should have been updated to look for a partial match of tags rather than an exact match.

**Comment by marko7460** [comment]
@aaron-lane i fixed the issue in this PR but if you want i can create a new PR just with that fix?

File: test/integration/node_pool/controls/kubectl.rb [hunk 2]
```diff
@@ -73,13 +73,13 @@
         end
 
         it "has the expected taints" do
-          expect(taints).to include(
+          expect(taints).to eq([
```
Comment (aaron-lane) [inline test/integration/node_pool/controls/kubectl.rb]: This appears to have been rebased incorrectly; #200 changed this to `include`.

Comment (marko7460) [reply test/integration/node_pool/controls/kubectl.rb]: no idea how that happened. It's fixed now.

**Pull request merged**